src/Advection/stretched_weno_smoothness.jl

```diff
@@ -0,0 +1,111 @@
+    return result
+end
+
+@inline left_biased_β₀(FT, ψ, T, scheme, args...) = biased_left_β(ψ, scheme, 0, args...) 
+@inline left_biased_β₁(FT, ψ, T, scheme, args...) = biased_left_β(ψ, scheme, 1, args...) 
+@inline left_biased_β₂(FT, ψ, T, scheme, args...) = biased_left_β(ψ, scheme, 2, args...) 
+
+@inline right_biased_β₀(FT, ψ, T, scheme, args...) = biased_right_β(ψ, scheme, 2, args...) 
+@inline right_biased_β₁(FT, ψ, T, scheme, args...) = biased_right_β(ψ, scheme, 1, args...) 
+@inline right_biased_β₂(FT, ψ, T, scheme, args...) = biased_right_β(ψ, scheme, 0, args...) 
+
+@inline retrieve_left_smooth(scheme, r, ::Val{1}, i, ::Type{Face})   = scheme.smooth_xᶠᵃᵃ[r+1][i] 
+@inline retrieve_left_smooth(scheme, r, ::Val{1}, i, ::Type{Center}) = scheme.smooth_xᶜᵃᵃ[r+1][i] 
+@inline retrieve_left_smooth(scheme, r, ::Val{2}, i, ::Type{Face})   = scheme.smooth_yᵃᶠᵃ[r+1][i] 
+@inline retrieve_left_smooth(scheme, r, ::Val{2}, i, ::Type{Center}) = scheme.smooth_yᵃᶜᵃ[r+1][i] 
+@inline retrieve_left_smooth(scheme, r, ::Val{3}, i, ::Type{Face})   = scheme.smooth_zᵃᵃᶠ[r+1][i] 
+@inline retrieve_left_smooth(scheme, r, ::Val{3}, i, ::Type{Center}) = scheme.smooth_zᵃᵃᶜ[r+1][i] 
+
+@inline retrieve_right_smooth(scheme, r, ::Val{1}, i, ::Type{Face})   = scheme.smooth_xᶠᵃᵃ[r+4][i] 
+@inline retrieve_right_smooth(scheme, r, ::Val{1}, i, ::Type{Center}) = scheme.smooth_xᶜᵃᵃ[r+4][i] 
+@inline retrieve_right_smooth(scheme, r, ::Val{2}, i, ::Type{Face})   = scheme.smooth_yᵃᶠᵃ[r+4][i] 
+@inline retrieve_right_smooth(scheme, r, ::Val{2}, i, ::Type{Center}) = scheme.smooth_yᵃᶜᵃ[r+4][i] 
+@inline retrieve_right_smooth(scheme, r, ::Val{3}, i, ::Type{Face})   = scheme.smooth_zᵃᵃᶠ[r+4][i] 
+@inline retrieve_right_smooth(scheme, r, ::Val{3}, i, ::Type{Center}) = scheme.smooth_zᵃᵃᶜ[r+4][i] 
+
+@inline calc_smoothness_coefficients(FT, ::Val{false}, args...; kwargs...) = nothing
+@inline calc_smoothness_coefficients(FT, ::Val{true}, coord::OffsetArray{<:Any, <:Any, <:AbstractRange}, arch, N; order) = nothing
+@inline calc_smoothness_coefficients(FT, ::Val{true}, coord::AbstractRange, arch, N; order) = nothing
+
+function calc_smoothness_coefficients(FT, beta, coord, arch, N; order) 
+
+    cpu_coord = arch_array(CPU(), coord)
+
+    order == 3 || throw(ArgumentError("The stretched smoothness coefficients are only implemented for order == 3"))
+    
+    s1 = create_smoothness_coefficients(FT, 0, -, cpu_coord, arch, N; order)
+    s2 = create_smoothness_coefficients(FT, 1, -, cpu_coord, arch, N; order)
+    s3 = create_smoothness_coefficients(FT, 2, -, cpu_coord, arch, N; order)
+    s4 = create_smoothness_coefficients(FT, 0, +, cpu_coord, arch, N; order)
+    s5 = create_smoothness_coefficients(FT, 1, +, cpu_coord, arch, N; order)
+    s6 = create_smoothness_coefficients(FT, 2, +, cpu_coord, arch, N; order)
+    
+    return (s1, s2, s3, s4, s5, s6)
+end
+
+function create_smoothness_coefficients(FT, r, op, cpu_coord, arch, N; order)
+
+    # derivation written on overleaf
```

Comment:
is there a public view link? add it?

Reply:
I second that, provided it's not too hard to do. Depending on how easy it is, it might be even worth writing it out quickly on the PR and then linking the comment here (now that github allows latex for math)